### PR TITLE
[Feat] Add resiliency to SSE client.

### PR DESCRIPTION
**Pull request by paul-amonson**
Signed-off-by: Paul Amonson <paul.d.amonson@intel.com>

### CODE REVIEW

File: authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java [hunk 1]
```diff
@@ -41,18 +41,24 @@ public void initialize(Logger logger, Map<String, String> arguments) throws Toke
...
-                .grantType(OAuth2Constants.CLIENT_CREDENTIALS);
-        supportRealm(arguments, builder);
-        supportUsernamePassword(arguments, builder);
-        log_.info("Creating an OAuth Token Bearer class for interface %s",
-                TokenAuthentication.class.getCanonicalName());
+                .realm(getRequiredArgumentValue(arguments, "realm"));
+        if(arguments.containsKey("username") && arguments.containsKey("password")) {
+            builder.grantType(OAuth2Constants.PASSWORD)
+                    .username(getRequiredArgumentValue(arguments, "username"))
+                    .password(getRequiredArgumentValue(arguments, "password"));
```
Comment (sindhu-106) [inline authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java:54]: In previous design, clientsecret/username/password are always set if exists, but now it is made of either of them. Is this expected as per design ?

File: fabric/src/main/java/com/intel/dai/fabric/FabricAdapter.java [hunk 2]
```diff
@@ -336,12 +324,11 @@ void badParameterValue(String parameter, String reason) throws AdapterException
         config.put("eventsTopic", "ucs_fabric_events");
 
         // Connection information...
-        config.put("use-ssl", "false");
```
Comment (sindhu-106) [inline fabric/src/main/java/com/intel/dai/fabric/FabricAdapter.java:339]: Are we not using ssl ?

File: authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java [hunk 3]
```diff
@@ -41,18 +41,24 @@ public void initialize(Logger logger, Map<String, String> arguments) throws Toke
...
         KeycloakBuilder builder = getBuilder()
                 .clientId(getRequiredArgumentValue(arguments, "clientId"))
-                .clientSecret(getRequiredArgumentValue(arguments, "clientSecret"))
                 .serverUrl(getRequiredArgumentValue(arguments, "tokenServer"))
-                .grantType(OAuth2Constants.CLIENT_CREDENTIALS);
-        supportRealm(arguments, builder);
-        supportUsernamePassword(arguments, builder);
-        log_.info("Creating an OAuth Token Bearer class for interface %s",
-                TokenAuthentication.class.getCanonicalName());
+                .realm(getRequiredArgumentValue(arguments, "realm"));
```
[the remark this answers is not in the document]
Comment (paul-amonson) [reply authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java:50]: Yes, realm is required after experimenting on the working system.

File: authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java (hunk 1 above)
Comment (paul-amonson) [reply authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java:54]: This is as keycloak is designed you need realm and either client/secret or username/password. Our current system only uses realm and client/secret. But the keycloak server can be configured to use the username/password as well.

File: fabric/src/main/java/com/intel/dai/fabric/FabricAdapter.java (hunk 2 above)
Comment (paul-amonson) [reply fabric/src/main/java/com/intel/dai/fabric/FabricAdapter.java:339]: We are. It is now automatically determined by using either http ot https in the fullUrl key.

File: authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java [hunk 4]
```diff
@@ -41,18 +41,24 @@ public void initialize(Logger logger, Map<String, String> arguments) throws Toke
         if(logger == null) throw new IllegalArgumentException("'logger' is null but not allowed to be null");
         if(arguments == null) throw new IllegalArgumentException("'arguments' is null but not allowed to be null");
         log_ = logger;
-        log_.debug("OAuth Bearer Token configuration values:");
+        log_.debug("OAuth Bearer Token configuration values: %s", "Unknown");
```
[the remark this answers is not in the document]
Comment (paul-amonson) [reply authentication/src/main/java/com/intel/authentication/KeycloakTokenAuthentication.java:44]: This is an error. Good catch! will post a fix.

**Pull request merged**